src/librustc/error_codes.rs

```diff
@@ -1207,6 +1207,51 @@ fn main() {
 ```
 "##,
 
+E0284: r##"
+This error occurs when the compiler is unable to unambiguously infer the
+return type of a function or method which is generic on return type, such
+as the `collect` method for `Iterator`s.
+
+For example:
+
+```compile_fail,E0284
+fn foo() -> Result<bool, ()> {
+    let results = [Ok(true), Ok(false), Err(())].iter().cloned();
+    let v : Vec<bool> = results.collect()?;
```

Comment:
Usually you don't put a space between the variable and the semicolon.

```rust
let v: Vec<bool> = results.collect()?;
...
let temp: Result<Vec<bool>, ()> = results.collect();
```

Example:

https://github.com/rust-lang/rust/pull/61220/files#diff-e6786747070ada32f620caf54905df92R1117

Reply:
OK changed it.